### PR TITLE
🌍 : Add Hebrew Translation

### PR DESCRIPTION
**Pull request by ido777**
## Summary

Add Hebrew translation 

## Change Type

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update
- [X] Documentation update
- [X] Translation update

## Testing

TEsted locally 

## Checklist

- [X] My code adheres to this project's style guidelines
- [X] I have performed a self-review of my own code
- [ ] I have commented in any complex areas of my code
- [X] I have made pertinent documentation changes
- [X] My changes do not introduce new warnings
- [ ] I have written tests demonstrating that my changes are effective or that my feature works
- [X] Local unit tests pass with my changes
- [ ] Any changes dependent on mine have been merged and published in downstream modules.

### CODE REVIEW

File: client/src/localization/languages/He.tsx [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+  com_nav_lang_russian: 'Русский',
+  com_nav_lang_japanese: '日本語',
+  com_nav_lang_swedish: 'Svenska',
+  com_nav_lang_korean: '한국어',
+  com_nav_lang_vietnamese: 'Tiếng Việt',
+  com_nav_lang_traditionalchinese: '繁體中文',
+  com_nav_lang_arabic: 'العربية',
+  com_nav_lang_turkish: 'Türkçe',
+  com_nav_lang_dutch: 'Nederlands',
+  com_nav_lang_indonesia: 'Indonesia',
```
Comment (berry-13) [inline client/src/localization/languages/He.tsx]: remove this

File: client/src/localization/Translation.tsx [hunk 2]
```diff
@@ -16,10 +16,12 @@ import Vietnamese from './languages/Vi';
 import Turkish from './languages/Tr';
 import Dutch from './languages/Nl';
 import Indonesia from './languages/Id';
+import Hebrew from './languages/He';
 
 // === import additional language files here === //
 
 const languageMap: { [key: string]: unknown } = {
+  'he-HE': Hebrew,
```
Comment (berry-13) [inline client/src/localization/Translation.tsx]: put this under the Indonesia one

File: client/src/components/Nav/SettingsTabs/General/General.tsx [hunk 3]
```diff
@@ -87,6 +87,7 @@ export const LangSelector = ({
   const languageOptions = [
     { value: 'auto', display: localize('com_nav_lang_auto') },
     { value: 'en-US', display: localize('com_nav_lang_english') },
+    { value: 'he-HE', display: localize('com_nav_lang_hebrew') },
```
Comment (berry-13) [inline client/src/components/Nav/SettingsTabs/General/General.tsx]: put this under the Indonesia one

File: client/src/localization/languages/Eng.tsx [hunk 4]
```diff
@@ -370,6 +370,7 @@ export default {
   com_nav_language: 'Language',
   com_nav_lang_auto: 'Auto detect',
   com_nav_lang_english: 'English',
+  com_nav_lang_hebrew: 'עברית',
```
Comment (berry-13) [inline client/src/localization/languages/Eng.tsx]: put this under the Indonesia one

**Review by berry-13 — CHANGES_REQUESTED** [review]

File: client/src/localization/languages/He.tsx (hunk 1 above)
Comment (ido777) [reply client/src/localization/languages/He.tsx]: Don't you think 
  com_nav_language: 'שפה',
 Is still needed? 
 I know on other systems that when the setup is in the language you don't know, understanding the language menu is crucial...

**Comment by ido777** [comment]
Well, based on your comments, (and my initial version...) I started to think about the following.
 
There are actually 2 different needs for order. 
1) For a developer the order should be the easiest for developing needs.  In other words, in a key-value pairs file like the above, it might be sorted keys. 
2) For the end user the only order that matters is the UI, where sorted keys might work but a better one will probably be automatic language detection or common languages first.

AFAIU your criteria for the order is the order of adding to the code base, it is correct? however, this criteria while being logical to keep the history, is less appealing to a newcomer :-), and might not work for the long run.

What do you think?

**Comment by Berry-13** [comment]
> Well, based on your comments, (and my initial version...) I started to think about the following.
> 
>  
> 
> There are actually 2 different needs for order. 
> 
> 1) For a developer the order should be the easiest for developing needs.  In other words, in a key-value pairs file like the above, it might be sorted keys. 
> 
> 2) For the end user the only order that matters is the UI, where sorted keys might work but a better one will probably be automatic language detection or common languages first.
> 
> 
> 
> AFAIU your criteria for the order is the order of adding to the code base, it is correct? however, this criteria while being logical to keep the history, is less appealing to a newcomer :-), and might not work for the long run.
> 
> 
> 
> What do you think?

Yes, there isn’t an official criteria for this yet. We’re considering automation due to numerous missing translations. For now, either move them down or leave them as is. Place your language in the general settings at the bottom, as it’s not a commonly used language. I’ll consider a more organized approach later. Thank you for your suggestions

**Comment by ido777** [comment]
I already changed the code per what you asked, it was only thoughts...
The only difference between us is the line 
com_nav_language: 'שפה',
which you suggested to erase - and I asked about it above,  So you want me to erase it?

File: client/src/localization/languages/He.tsx (hunk 1 above)
Comment (berry-13) [reply client/src/localization/languages/He.tsx]: exact, that's why language should be only in English. If someone has some strange bugs and doesn't understand the language that Librechat is set to, he approximately needs to know where to change it. The setting button has an icon, but the language label doesn't have it

Comment (ido777) [reply client/src/localization/languages/He.tsx]: You are right
Will fix it later

**Review by berry-13 — APPROVED** [review]
thank you!

**Pull request merged**